rtos/EventFlags.h

```diff
@@ -0,0 +1,99 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#ifndef EVENT_FLAG_H
+#define EVENT_FLAG_H
+
+#include <stdint.h>
+#include "cmsis_os2.h"
+#include "mbed_rtos1_types.h"
+#include "mbed_rtos_storage.h"
+
+#include "platform/NonCopyable.h"
+
+namespace rtos {
+/** \addtogroup rtos */
+/** @{*/
+
+/** The EventFlags class is used to signal or wait for an arbitrary event or events.
+ @note 
+ EventFlags support 31 flags so the MSB flag is ignored, it is used to return an error code (@a osFlagsError)
+ @note
+ Memory considerations: The EventFlags control structures will be created on current thread's stack, both for the mbed OS
+ and underlying RTOS objects (static or dynamic RTOS memory pools are not being used).
+*/
+class EventFlags : private mbed::NonCopyable<EventFlags> {
+public:
+    /** Create and Initialize a EventFlags object */
+    EventFlags();
+
+    /** Create and Initialize a EventFlags object
+
+     @param name name to be used for this EventFlags. It has to stay allocated for the lifetime of the thread.
+    */
+    EventFlags(const char *name);
+
+    /** Set the specified Event Flags.
+      @param   flags  specifies the flags that shall be set.
+      @return  event flags after setting or error code if highest bit set (@a osFlagsError).
+     */
+    uint32_t set(uint32_t flags);
+
+    /** Clear the specified Event Flags.
+      @param   flags  specifies the flags that shall be cleared. (default: 0x7fffffff - all flags)
+      @return  event flags before clearing or error code if highest bit set (@a osFlagsError).
+     */
+    uint32_t clear(uint32_t flags = 0x7fffffff);
+
+    /** Get the currently set Event Flags.
+      @return  set event flags.
+     */
+    uint32_t get();
+
+    /** Wait for all of the specified event flags to become signaled.
+      @param   flags    specifies the flags to wait for.
+      @param   timeout  timeout value or 0 in case of no time-out. (default: osWaitForever)
+      @return  event flags before clearing or error code if highest bit set (@a osFlagsError).
+      @note    flags that are specified to wait for will be automatically cleared.
+     */
+    uint32_t wait_all(uint32_t flags = 0, uint32_t timeout = osWaitForever);
```

Comment:
You don't. I was thinking that we had issues with the RTOS APIs changing and exposing 95% of functionality in abstracted out, simpler and more intuitive way is better than having C++ shim on rtx object. Do we have a use case that `osFlagsNoClear` is needed and can't be achieved another way?